### PR TITLE
fix: dismiss to conversations on "Back To Wire" and "Go To Team Management - WPB-15082

### PR DESCRIPTION
**Pull request by El-Fitz**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-15082" title="WPB-15082" target="_blank">BugWPB-15082</a>  [iOS] Back to wire option after creating a team is not taking user to conversations
  </td></table>
  <br />
 

### Issue

When completing the individual to team migration flow, on selecting either "Back To Wire" or "Go To Team Management",  the current modal would be dismissed and, if the user had tapped on "Go To Team Management", the appropriate URL would be opened using the browser.

Instead, when selecting either "Back To Wire" or "Go To Team Management", both the current and the profile's modals should be dismissed (before opening the appropriate browser if the user has tapped on "Go To Team Management").

### Testing

Sign up on staging using a personal account with v7 as the preferred API (when v7 is enabled in staging).
Go through the personal to teams flows.
At the end, tap on "Back To Wire". Both the current and the profile's modals should be dismissed.

Sign up on staging using a personal account with v7 as the preferred API (when v7 is enabled in staging).
Go through the personal to teams flows.
At the end, tap on "Go To Team Management". Both the current and the profile's modals should be dismissed. The team management URL should be opened in your selected browser.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.

### CODE REVIEW

**Review by jullianm — APPROVED** [review]
LGTM

**Review by netbe — APPROVED** [review]
looks good

**Comment by El-Fitz** [comment]
Pending feedback on analytics (see `IndividualToTeamMigrationViewController.swift:242`)

**Comment by El-Fitz** [comment]
~~The way analytics are currently implemented, `user.personal-team-creation-flow-completed` won't be tracked if the user dismisses the modal in the last step by dragging the view.~~ fixed

File: WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift [hunk 1]
```diff
@@ -408,3 +421,4 @@ private func viewFor(
 #Preview {
     featurePreview()
 }
+Í
```
Comment (netbe) [inline WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift]: typo?

File: WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent+User.swift [hunk 2]
```diff
@@ -90,17 +90,19 @@ public extension AnalyticsEvent {
...
+            case .none:
+                nil
+            case .backToWire:
+                SegmentationEntry(key: "modal_back-to-wire-clicked", value: true)
+            case .openTeamManagement:
+                SegmentationEntry(key: "modal_open-tm-clicked", value: true)
+            }
+            return AnalyticsEvent(name: "user.personal-team-creation-flow-completed") {
+                if let segment { segment }
             }
```
Comment (netbe) [inline WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent+User.swift]: question: I don't understand the syntax here, what's the closure taking? I thought it was a set of SegmentationEntry

File: WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift (hunk 1 above)
Comment (netbe) [inline WireUI/Sources/WireIndividualToTeamMigrationUI/IndividualToTeamMigrationViewController.swift]: typo?

**Review by netbe — CHANGES_REQUESTED** [review]
one character is causing compilation error

File: WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent+User.swift (hunk 2 above)
Comment (El-Fitz) [reply WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent+User.swift]: It's a Result Builder

**Review by netbe — APPROVED** [review]
LGTM;)

**Pull request merged**